tests/functional/backend/SettingsPluginTest.php

```diff
@@ -147,12 +189,143 @@ public function testGetAndSetSettingEncrypted()
     }
 
     /**
+     * Test that date time settings are saved using a
+     * format specified in datetimesaveformat or if
+     * not set, they are saved using
+     * the session date format.
+     */
+    public function testGetAndSetDateTimeSettings(): void
+    {
+        self::$plugin->setSettings(self::$dateTimePluginSettings);
+        $sessionFormatDate = getDateFormatData(App()->session['dateformat']);
+        /* format sent by widget */
+        $sendformat = $sessionFormatDate['phpdate'] . ' H:i';
+        /* Check with each settings with current datetime */
+        foreach (self::$dateTimePluginSettings as $pluginSetting => $settingDatas) {
+            /* Set the value to current date */
+            $value = date_create()->format($sendformat);
+            self::$plugin->setSetting($pluginSetting, $value);
+
+            $setting = \PluginSetting::model()->findByAttributes([
+                'plugin_id' => self::$plugin->getId(),
+                'key' => $pluginSetting
+            ]);
+
+            $format = !empty($settingDatas['saveformat']) ? $settingDatas['saveformat'] : $sessionFormatDate['phpdate'] . ' H:i';
+            $date = \LimeSurvey\PluginManager\LimesurveyApi::getFormattedDateTime($value, $format);
+
+            $this->assertNotEmpty(
+                $setting->id,
+                'The setting id is empty, there must be a problem while saving ' . $value
+            );
+            $this->assertEquals(
+                $setting->value,
+                json_encode($date),
+                'The value returned in the PluginSetting object after saving is not correct for ' . $pluginSetting . '.'
+            );
+
+            $settingValue = self::$plugin->getSetting($pluginSetting);
+            $this->assertEquals(
+                $settingValue,
+                $date,
+                'The value returned by the get function is not correct for ' . $pluginSetting . '.'
+            );
+        }
+    }
+
+    public function testSaveDateTimePluginSettingWithoutASpecificFormat()
```

Comment:
just see it's already done on testGetAndSetDateTimeSettings

Reply:
Are you commenting on your own PR? :) You think this test can be deleted?